aea/aea.py

```diff
@@ -53,11 +54,12 @@ def __init__(self, name: str,
         :param mailbox: the mailbox of the agent.
         :param wallet: the wallet of the agent.
         :param ledger_apis: the ledger apis of the agent.
+        :param resources: the resources of the agent. If None, the directory parameter will be considered.
+        :param directory: the path to the agent's resource directory.
```

Comment:
Why are we adding a path if we pass in resources? This is rhetoric, as I know why :). Seems messy. I think we could do the following: instantiate `resources` and make it default requirement. But instantiate means it just has the path as part of the object constructor, no loading of resources at this point. Then give it a `load` method which actually populates the registry.